runtime/io.c

```diff
@@ -1035,5 +1043,8 @@ CAMLprim value caml_ml_input_scan_line(value vchannel)
 
 CAMLprim value caml_terminfo_rows(value vchannel)
 {
-  return Val_int(caml_num_rows_fd(Channel(vchannel)->fd));
+  CAMLparam1 (vchannel);
+  struct channel * channel = Channel(vchannel);
+  int num = caml_num_rows_fd(channel->fd);
+  CAMLreturn (Val_int(num));
```

Comment:
I believe the original code is safe and this change is not necessary.  "But it doesn't hurt!" you're saying?  I prefer bug-fix commit that contain only code strictly necessary to fix the bug.

Reply:
Ok, I remove the root-registration here.

(`caml_num_rows_fd` does not currently call the GC in either its Unix and Windows versions. This is not documented and might change in the future, so I thought that it was more robust to root anyway.)